module/renderer/src/commonMain/kotlin/org/cru/godtools/shared/renderer/lesson/RenderLesson.kt

```diff
@@ -102,8 +107,18 @@ fun RenderLesson(state: LessonScreen.UiState, modifier: Modifier = Modifier) {
                         }
                     },
                     actions = {
-                        // HACK: to center the title, we add a spacer the same size as the navigation icon
-                        Spacer(modifier = Modifier.width(48.dp))
+                        if (state is LessonScreen.UiState.Loaded && state.manifest.code != null && state.manifest.locale != null) {
```

Comment:
let's add a `showShareAction` boolean property to the `UiState` object. it can default to false.

We are trying to separate the business logic from the UI where possible, eventually we will be creating a Circuit `Presenter` in the android app to generate the `UiState` objects and handle any callbacks appropriately.